Mathlib/Topology/Algebra/ConstMulAction.lean

```diff
@@ -474,19 +494,19 @@ section MulAction
 variable {G₀ : Type*} [GroupWithZero G₀] [MulAction G₀ α] [TopologicalSpace α]
   [ContinuousConstSMul G₀ α]
 
--- Porting note: generalize to a group action + `IsUnit`
-/-- Scalar multiplication preserves neighborhoods. -/
-theorem set_smul_mem_nhds_smul {c : G₀} {s : Set α} {x : α} (hs : s ∈ 𝓝 x) (hc : c ≠ 0) :
-    c • s ∈ 𝓝 (c • x : α) := by
-  rw [mem_nhds_iff] at hs ⊢
-  obtain ⟨U, hs', hU, hU'⟩ := hs
-  exact ⟨c • U, Set.smul_set_mono hs', hU.smul₀ hc, Set.smul_mem_smul_set hU'⟩
+-- Porting note: generalize to a monoid action + `IsUnit`
+/-- Scalar multiplication by a nonzero scalar preserves neighborhoods. -/
+theorem smul_mem_nhds_smul₀ {c : G₀} {s : Set α} {x : α} (hc : c ≠ 0) :
+    c • s ∈ 𝓝 (c • x : α) ↔ s ∈ 𝓝 x :=
+  smul_mem_nhds_smul (Units.mk0 c hc)
 
-theorem set_smul_mem_nhds_smul_iff {c : G₀} {s : Set α} {x : α} (hc : c ≠ 0) :
-    c • s ∈ 𝓝 (c • x : α) ↔ s ∈ 𝓝 x := by
-  refine ⟨fun h => ?_, fun h => set_smul_mem_nhds_smul h hc⟩
-  rw [← inv_smul_smul₀ hc x, ← inv_smul_smul₀ hc s]
-  exact set_smul_mem_nhds_smul h (inv_ne_zero hc)
+@[deprecated (since := "2024-08-06")]
+alias set_smul_mem_nhds_smul_iff := smul_mem_nhds_smul₀
+
+@[deprecated  smul_mem_nhds_smul₀ (since := "2024-08-06")]
+theorem set_smul_mem_nhds_smul {c : G₀} {s : Set α} {x : α} (hs : s ∈ 𝓝 x) (hc : c ≠ 0) :
+    c • s ∈ 𝓝 (c • x : α) :=
+  (smul_mem_nhds_smul₀ hc).2 hs
```

Comment:
can we just deprecate this, but leave an undeprecated alias with the name `smul_mem_nhds₀`?